src/rgw/rgw_process.cc

```diff
@@ -119,8 +119,7 @@ int process_request(RGWRados* const store,
                     int* http_ret)
 {
   int ret = 0;
-
-  client_io->init(g_ceph_context);
+  int cio_error = client_io->init(g_ceph_context);
```

Comment:
having 3 different variables (ret, cio_error, init_error) for error handling in this function is a bit much. can you use ret here, and add a new `abort_early()` check as soon as the req_state is initialized?

that way, we won't call `rest->get_handler()` unnecessarily, and the error handling for that call doesn't need to juggle the different return codes, ie. `init_error ? init_error : cio_error`

Reply:
I tried going that path before, If we return after just initializing rstate and return we assert later as s->cio is unset and abort_early uses that when dumping status.

Reply:
okay, makes sense

Reply:
Let me try setting cio before calling abort